### PR TITLE
Grades bug solved

### PR DESCRIPTION
**Pull request by AlejandroAdrian**
Signed-off-by: Alejandro <alejandro.adrian@exponentiateam.com>

### CODE REVIEW

File: app/controllers/report_controller.rb [hunk 1]
```diff
@@ -1,14 +1,15 @@
 class ReportController < AuthenticatedController
   def show
-    report = find_report
-
-    if report.present?
-      pdf = Base64.decode64(report.content)
-      send_data(pdf, filename: report.filename)
-      return
+    if find_course_mark.present?
```
Comment (mapreal19) [inline app/controllers/report_controller.rb:3]: I'd use a guard clause:

```ruby
if find_course.mark.blank?
  redirect_to grades_path, alert: "Must put marks before generating the report."
  return
end
```

File: app/controllers/report_controller.rb [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+      if find_report.present?
+        pdf = Base64.decode64(find_report.content)
+        send_data(pdf, filename: find_report.filename)
+        return
+      end
+      create_report_with_pdf
+    else
+      redirect_to grades_path, alert: "Must put marks before generating the report."
```
Comment (mapreal19) [inline app/controllers/report_controller.rb:11]: use i18n. And maybe a better message: "can't generate the report without grades"?

File: spec/requests/report_request_spec.rb [hunk 3]
```diff
@@ -33,5 +33,20 @@
         expect(Report.count).to eq(1)
       end
     end
+
+    context "when try to generate report without course mark" do
+      it "doesn't generate report", :vcr do
```
Comment (mapreal19) [inline spec/requests/report_request_spec.rb:38]: don't need `vcr` in this one

File: spec/requests/report_request_spec.rb [hunk 4]
```diff
@@ -33,5 +33,20 @@
         expect(Report.count).to eq(1)
       end
     end
+
+    context "when try to generate report without course mark" do
```
Comment (mapreal19) [inline spec/requests/report_request_spec.rb:37]:
```suggestion
    context "when generating a report without marks" do
```

File: spec/requests/report_request_spec.rb [hunk 5]
```diff
@@ -33,5 +33,20 @@
...
+        director = create(:user, :director)
+        sign_in(director)
+        course = create(:course)
+        student = create(:student)
+        student.courses << course
+
+        get "/report/#{student.id}.pdf?course_id=#{course.id}"
+
+        expect(response).to have_http_status(:redirect)
+        expect(flash[:alert]).to eq("Must put marks before generating the report.")
```
Comment (mapreal19) [inline spec/requests/report_request_spec.rb:48]: I'd add another expect to ensure we're not creating a report in this case:

`expect(Report.count).to eq(0)`

**Review by mapreal19 — CHANGES_REQUESTED** [review]

File: app/controllers/report_controller.rb (hunk 1 above)
Comment (AlejandroAdrian) [reply app/controllers/report_controller.rb:3]: Done

File: app/controllers/report_controller.rb (hunk 2 above)
Comment (AlejandroAdrian) [reply app/controllers/report_controller.rb:11]: Done

File: spec/requests/report_request_spec.rb (hunk 3 above)
Comment (AlejandroAdrian) [reply spec/requests/report_request_spec.rb:38]: Done

File: spec/requests/report_request_spec.rb (hunk 4 above)
Comment (AlejandroAdrian) [reply spec/requests/report_request_spec.rb:37]: Done

File: spec/requests/report_request_spec.rb (hunk 5 above)
Comment (AlejandroAdrian) [reply spec/requests/report_request_spec.rb:48]: Done

**Review by mapreal19 — APPROVED** [review]

**Pull request merged**